staging/src/k8s.io/client-go/dynamic/dynamicinformer/informer.go

```diff
@@ -138,20 +152,65 @@ func NewFilteredDynamicInformer(client dynamic.Interface, gvr schema.GroupVersio
 			resyncPeriod,
 			indexers,
 		),
+		gvr:     gvr,
+		stopCh:  stopCh,
+		factory: factory,
 	}
 }
 
 type dynamicInformer struct {
-	informer cache.SharedIndexInformer
-	gvr      schema.GroupVersionResource
+	cache.SharedIndexInformer
+	gvr    schema.GroupVersionResource
+	stopCh chan struct{}
+	// once is for safely close stopCh
+	once    sync.Once
+	factory *dynamicSharedInformerFactory
 }
 
 var _ informers.GenericInformer = &dynamicInformer{}
 
 func (d *dynamicInformer) Informer() cache.SharedIndexInformer {
-	return d.informer
+	return d
 }
 
 func (d *dynamicInformer) Lister() cache.GenericLister {
-	return dynamiclister.NewRuntimeObjectShim(dynamiclister.New(d.informer.GetIndexer(), d.gvr))
+	return dynamiclister.NewRuntimeObjectShim(dynamiclister.New(d.GetIndexer(), d.gvr))
+}
+
+var _ cache.SharedIndexInformer = &dynamicInformer{}
+
+func (d *dynamicInformer) GetController() cache.Controller {
+	return &dummyController{informer: d}
+}
+
+// Use local stopCh, the passed in stopCh is ignored.
+// The informer is stopped via its controller
+func (d *dynamicInformer) Run(stopCh <-chan struct{}) {
+	d.SharedIndexInformer.Run(d.stopCh)
```

Comment:
d.Run will be called multiple times as it's a shared dynamic informer. Everytime, it calls to d.SharedIndexInformer.Run, which will [create](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/cache/shared_informer.go#L261) a new concrete cache.controller with reflector and cache. This defeats the purpose of a shared informer.